src/hotspot/share/runtime/deoptimization.cpp

```diff
@@ -440,8 +440,19 @@ static void restore_eliminated_locks(JavaThread* thread, GrowableArray<compiledV
     }
   }
 #ifdef ASSERT
-  if (LockingMode == LM_LIGHTWEIGHT && !realloc_failures) {
-    deoptee_thread->lock_stack().verify_consistent_lock_order(lock_order, exec_mode != Deoptimization::Unpack_none);
+  if (lock_order.is_nonempty()) {
+    const int bci = chunk->first()->bci();
+    const Bytecodes::Code bc = chunk->first()->method()->code_at(bci);
+    const bool previous_bc_not_monitorenter = bci == 0 ? !chunk->first()->method()->is_synchronized()
+                                                       : chunk->first()->method()->code_at(bci - 1) != Bytecodes::_monitorenter;
+    const bool is_syncronized_entry = chunk->first()->method()->is_synchronized() &&
+                                      chunk->first()->raw_bci() == SynchronizationEntryBCI;
+    // If deoptimizing from monitorenter bytecode we maybe in transitional state. Skip verification.
```

Comment:
nit typo: s/we maybe/we may be/
nit typo: s/in transitional state/is a transitional state/